force-app/main/default/classes/OpportunityUtils_Test.cls

```diff
@@ -7,4 +7,110 @@ private class OpportunityUtils_Test
     // so don't stop just because you have full coverage.
+
+
+    //Tests all scenarios for a variety of amounts in 200 records
+    @isTest
+    private static void testBulk()
+    {
+        User salesUser = UnitTestHelper.CreateUser_inMemory('Joe', 'Bloggs', 'jbl', 'Standard User');
+        insert salesUser;
+
+        Account testAcc = UnitTestHelper.createAccount_InMemory();
+        insert testAcc;
+
+        //Creates 200 opps with amounts 1k to 200k
+        List<Opportunity> testOpps = UnitTestHelper.createOpps_InMemory(testAcc, 200);
+
+        Test.StartTest();
+        System.runAs(salesUser)
+        {
+            OpportunityUtils oppUtils = new OpportunityUtils();
+            oppUtils.addOpportunitySizeToNameAndSave(testOpps); 
+            //Why isn't this a static method?
+        }
+        Test.StopTest();
+
+        List<Opportunity> resultOpps = [SELECT Id, Name, Amount 
+                                        FROM Opportunity 
+                                        WHERE Id IN: testOpps
+                                        ORDER BY Amount];
+
+        System.assertEquals(200, resultOpps.size(), 'Expected 200 test opps');
+
+        Opportunity opp = new Opportunity();
+
+        for (Integer i = 0; i < 50; i++) 
+        {
+            opp = resultOpps[i];
+            System.assert(opp.Amount <= 50000, 'Testing should be for small amounts');
+            System.assertEquals(true, opp.Name.endsWith('Small'), 'Wrong opp size in name');
+            System.assertNotEquals(true, opp.Name.endsWith('Medium'), 'Wrong opp size in name');
+            System.assertNotEquals(true, opp.Name.endsWith('Large'), 'Wrong opp size in name');
+            System.assert(opp.Name.endsWith('Small'), 'Wrong opp size in name');
+        }
+
+        for (Integer i = 50; i < 100; i++) 
+        {
+            opp = resultOpps[i];
+            System.assert(opp.Amount > 50000, 'Testing should be for Medium amounts');
+            System.assert(opp.Amount <= 100000, 'Testing should be for Medium amounts');
+            System.assertEquals(true, opp.Name.endsWith('Medium'), 'Wrong opp size in name');
+            System.assertNotEquals(true, opp.Name.endsWith('Small'), 'Wrong opp size in name');
+            System.assertNotEquals(true, opp.Name.endsWith('Large'), 'Wrong opp size in name');
+
+
+        }
+
+        for (Integer i = 100; i < 200; i++) 
+        {
+            opp = resultOpps[i];
+            System.assert(opp.Amount > 100000, 'Testing should be for Large amounts');
+            System.assertEquals(true, opp.Name.endsWith('Large'), 'Wrong opp size in name');
+            System.assertNotEquals(true, opp.Name.endsWith('Small'), 'Wrong opp size in name');
+            System.assertNotEquals(true, opp.Name.endsWith('Medium'), 'Wrong opp size in name');
+        }
+    }
+
+    //test boundaries
+    //0
+    //50000
+    //100000
+
+    @IsTest
+    private static void testBoundaryAmounts()
```

Comment:
From a best practice perspective, I would break this up into three separate tests. Right now, if the first opportunity name doesn't end with 'Small', you will receive an error and the test will stop. At this point you don't know if any of it works, or if it is just the small case that has been broken. If you have a separate test for each of small, medium and large, you will know which is working and which isn't.